app/models/miq_queue.rb

```diff
@@ -486,6 +486,14 @@ def deliver(requester = nil, &block)
     return status, message, result
   end
 
+  # @return true if completed, false if timed out
+  def deliver_and_process(requester = nil, &block)
+    status, message, result = deliver(requester = nil, &block)
+    delivered(status, message, result) unless status == STATUS_RETRY
+
+    status != MiqQueue::STATUS_TIMEOUT
```

Comment:
I don't see this return value being used anywhere, but I think it would be more useful to return the `status` directly.  The caller, if it needs to check it, can check against `MiqQueue::STATUS_TIMEOUT`